pkg/endpoint/bpf.go

```diff
@@ -154,13 +159,17 @@ func (e *Endpoint) writeHeaderfile(prefix string) error {
 		oldHeaderPath := filepath.Join(prefix, common.OldCHeaderFileName)
 		if _, err := os.Stat(oldHeaderPath); err != nil {
 			// The symlink doesn't already exists.
-			if err := os.Symlink(common.CHeaderFileName, oldHeaderPath); err != nil {
+			if err := renameio.Symlink(common.CHeaderFileName, oldHeaderPath); err != nil {
```

Comment:
This will create the symlink even if we later fail to atomically replace the target file, no?

Reply:
Right, I was assuming it would then link to the old, existing file. But this assumption of course isn't true if no header file exists yet during initial creation. Thanks for pointing this out, should now be accounted for as well.